{"version":3,"sources":["../../app-src/util/ApplicationException.js"],"names":["isApplicationException","err","exception","Object","getPrototypeOf","getExceptionMessage","message","console","log","ApplicationException","msg","name","constructor","Error"],"mappings":";;;;;;;;;;;;;;;;;;;;;;;;;;;;;AAWO,WAASA,sBAAT,CAAgCC,GAAhC,EAAqC;AAExC,WAAOA,GAAG,YAAYC,SAAf,IAA4BC,MAAM,CAACC,cAAP,CAAsBH,GAAtB,aAAsCC,SAAzE;AACH;;AAEM,WAASG,mBAAT,CAA6BJ,GAA7B,EAAkC;AAErC,QAAGD,sBAAsB,CAACC,GAAD,CAAzB,EAAgC;AAC5B,aAAOA,GAAG,CAACK,OAAX;AACH,KAFD,MAEO;AACHC,MAAAA,OAAO,CAACC,GAAR,CAAYP,GAAZ;AACA,aAAO,uCAAP;AACH;AACJ;;;4BAbeD,sB;yBAKAK;;;;;;sCAhBHI,oB;;;;;AAET,wCAAsB;AAAA;;AAAA,cAAVC,GAAU,uEAAJ,EAAI;;AAAA;;AAElB,oGAAMA,GAAN;AACA,gBAAKC,IAAL,GAAY,MAAKC,WAAL,CAAiBD,IAA7B;AAHkB;AAIrB;;;yBANqCE,K;;AASpCX,MAAAA,S,GAAYO,oB","sourcesContent":["export class ApplicationException extends Error {\n\n    constructor(msg = '') {\n\n        super(msg);\n        this.name = this.constructor.name;\n    }\n} \n\nconst exception = ApplicationException;\n\nexport function isApplicationException(err) {\n\n    return err instanceof exception || Object.getPrototypeOf(err) instanceof exception \n}\n\nexport function getExceptionMessage(err) {\n    \n    if(isApplicationException(err)) {\n        return err.message;\n    } else {\n        console.log(err);\n        return 'Não foi possível realizar a operação.';\n    }\n}"],"file":"ApplicationException.js"}